File: prisma/refs/java/partition-equal-subset-sum.java
class Solution {
    public boolean canPartition(int[] nums) {
        int total = 0;
        for (int n : nums) total += n;
        if (total % 2 == 1) return false;
        int target = total / 2;
        boolean[] dp = new boolean[target + 1];
        dp[0] = true;
        for (int n : nums)
            for (int s = target; s >= n; s--)
                if (dp[s - n]) dp[s] = true;
        return dp[target];
    }
}

File: prisma/refs/py/partition-equal-subset-sum.py
def can_partition(nums):
    total = sum(nums)
    if total % 2:
        return False
    target = total // 2
    reachable = {0}
    for n in nums:
        reachable |= {s + n for s in reachable if s + n <= target}
        if target in reachable:
            return True
    return target in reachable
